File: runtime/cpp/cyana-ast/SyntaxDfaState.cpp
//
// Created by tian wei jun on 2022/12/1 0001.
//

#include "SyntaxDfaState.h"

SyntaxDfaState::SyntaxDfaState() : type(0),
                                   edges(std::map<Grammar *, SyntaxDfaState *, SyntaxDfaStateEdgesCompare>()) {
}

File: runtime/java/cyana-ast-runtime/src/cyana/ast/runtime/SyntaxDfaState.java
package cyana.ast.runtime;

import java.util.LinkedHashMap;
import java.util.LinkedList;
import java.util.Map;

/**
 * 语法对应的 确定有限状态自动机 的状态.
 *
 * @author tian wei jun
 */
public class SyntaxDfaState {
  public int type = FaStateType.NONE.getValue();
  // 转移
  public Map<Grammar, SyntaxDfaState> edges = new LinkedHashMap<>();
  public LinkedList<ProductionRule> closingProductionRules = new LinkedList<>();
}
